src/dmd/mars.d

```diff
@@ -360,6 +360,11 @@ private int tryMain(size_t argc, const(char)** argv)
     }
     if (files.dim == 0)
     {
+        if (global.params.jsonQuery)
+        {
+            generateJson(null);
+            return EXIT_SUCCESS;
```

Comment:
Ah nice. I didn't see this initially. 

This is more or less what I am trying to achieve in https://github.com/dlang/dmd/pull/7840